projects/lib/src/channel/channel-message-handler.ts

```diff
@@ -862,4 +862,103 @@ export class ChannelMessageHandler {
             );
         }
     }
+
+    /**
+     * Clean up all channel subscriptions for a disconnected proxy
+     * @param appId The app ID of the disconnected proxy
+     */
+    public cleanupDisconnectedProxy(appId: FullyQualifiedAppIdentifier): void {
```

Comment:
same here. This function is doping quite a lot of different things. As close as possible to one thing per function would be great! 😄